modules/standard/MemMove.chpl

```diff
@@ -67,11 +89,31 @@ module MemMove {
 
     :arg: A variable to deinitialize
   */
+  pragma "no doc"
+  deprecated "'explicitDeinit' is now deprecated; please use 'destroy' instead"
   proc explicitDeinit(ref arg: ?t) {
     if needsDeinit(t) then
       chpl__autoDestroy(arg);
   }
 
+  /*
+    Explicitly destroy a variable as the compiler would when its lifetime ends.
+    The variable referred to by ``obj`` should be considered unusable after a
+    call to this function, and its particular state is undefined.
```

Comment:
This should say something about taking no action for types where `needsDestroy` returns false.